### ISSUE TITLE
[a11y]: TabPanel code to set tabIndex doesn't work

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#19083]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.80.0

### React version

20

### Automated testing tool and ruleset

accessibility-checker

### Description

TabPanel has a bunch of code to set the tabIndex based on whether the content is interactive or not, including this main block:

```ts
useEffect(() => {
  if (!panel.current) {
    return;
  }

  const content = getInteractiveContent(panel.current);
  if (content) {
    setInteractiveContent(true);
    setTabIndex(-1);
  }
}, []);

// tabindex should only be 0 if no interactive content in children
useEffect(() => {
  const node = panel.current;
  if (!node) {
    return;
  }

  function callback() {
    const content = getInteractiveContent(node as HTMLElement);
    if (content) {
      setInteractiveContent(true);
      setTabIndex(-1);
    } else {
      setInteractiveContent(false);
      setTabIndex(0);
    }
  }
  const observer = new MutationObserver(callback);

  observer.observe(node, {
    childList: true,
    subtree: true,
  });
  return () => observer.disconnect();
}, []);
```

However, if you look at https://react.carbondesignsystem.com/?path=/story/components-tabs--contained, you can see that that tabIndex is always -1, both for tabs with and without interactive content:

<img width="990" alt="Image" src="https://github.com/user-attachments/assets/15b11589-c74c-46ce-a196-f1fdcfd85c6e" />

### Solution

I suggest to fix that by just removing all that code, as I did in #19405, although I understand that's a bit controversial because you're worried about supporting Mac Safari.

### WCAG 2.1 Violation

https://www.w3.org/WAI/standards-guidelines/act/rules/0ssw9k/proposed/

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tabs--contained

### Steps to reproduce

Just look in Chrome developer tools.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by adamalston** [comment]
Was the problematic `tabIndex` logic for `TabPanel` removed in https://github.com/carbon-design-system/carbon/pull/19405? If so, what's left to do here?

**Comment by wkeese** [comment]
Agreed, closing this, and updated the comment in #19405.
